### ISSUE TITLE
Support HOCs that wrap & export component

### ISSUE THREAD

**Issue by ben-seed** [storybookjs/babel-plugin-react-docgen#48]
No documentation is produced if a component is wrapped before being exported, e.g. using redux connect.

I've tried using different resolvers but this doesn't fix the problem. I can see in the react-docgen codebase history that [they have added support for HOCs](https://github.com/reactjs/react-docgen/pull/124), so I believe this limitation must be specific to this plugin.

**Comment by danielduan** [comment]
can you paste a full file that I can test? we do support some HOCs, but definitely think we missed some edge cases.

**Comment by ben-seed** [comment]
TestComponent.jsx:

`import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import testSelector from './testSelector';

class TestComponent extends React.Component {
    static propTypes = {
        /** Text to display */
        text: PropTypes.string,
        /** Called on click */
        onClick: PropTypes.func
    };

    static defaultProps = {
        text: 'hello'
    };

    render() {
        const { text, onClick } = this.props;

        return (
            <div onClick={onClick}>{text}</div>
        );
    }
}

export default connect(testSelector)(TestComponent);
`

testSelector.js:
`
export default () => 'hello';
`

**Comment by ben-seed** [comment]
TestComponent.jsx:
```
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { mapStateToProps, mapDispatchToProps } from './testSelector';

const TestComponent = (props) => {
    const { text, onClick } = props;
    
    return (
        <div>
            <div>Text: {text}</div>
            <button onClick={onClick}>Button</button>
        </div>
    );
};

TestComponent.propTypes = {
    /** Text to display */
    text: PropTypes.string,
    /** Called on click */
    onClick: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(TestComponent);
```

testSelector.js:
```
export const mapStateToProps = (state, props) => ({
    text: 'hello'
});

export const mapDispatchToProps = (dispatch) => ({
    onClick: () => dispatch({ type: 'ON_CLICK' })
});
```

**Comment by mattapperson** [comment]
Also running into this issue with similar code

**Comment by danielduan** [comment]
Fixed in v3.2.0
